src/compute/src/render/join/linear_join.rs

```diff
@@ -310,11 +338,17 @@ where
     T: Timestamp + Lattice,
     Tr1: TraceReader<Key = Row, Val = Row, Time = G::Timestamp, R = Diff> + Clone + 'static,
     Tr2: TraceReader<Key = Row, Val = Row, Time = G::Timestamp, R = Diff> + Clone + 'static,
+    MzArranged<G, Tr1>: ArrangementSize,
+    MzArranged<G, Tr2>: ArrangementSize,
 {
     // Reuseable allocation for unpacking.
     let mut datums = DatumVec::new();
     let mut row_builder = Row::default();
 
+    // Safety: all `join_impl`s holds on to the trace.
```

Comment:
```suggestion
    // Safety: all `join_impl`s hold on to the trace.
```

Reply:
The linear join impls drop their trace handles when the corresponding streams reach the empty frontier. I assume that's allowed under the safety requirements?

Reply:
Yes, that's the expected behavior. In this case, the trace should only contain a single empty batch.